### ISSUE TITLE
${logging_alert.messages_url} has a wrong timerange for a LESS THAN rule

### ISSUE THREAD

**Issue by frantz45** [airbus-cyber/graylog-plugin-logging-alert#47]
To reproduce (it's easier to create the rule via the Wizard but it can be created from Event definition)
1. Create a COUNT rule
2. Choose a condition that doesn't match any log you receive (for example field "aaaaaa" matches exactly "aaaaaa")
3. Select the count condition LESS THAN 1
4. Set the timerange condition to the last 10 minutes and Save the rule
5. Go the corresponding Event Definition and set the "Execute search every" to 10 minutes (so the rule is checked every 10 minutes on a window of the last 10 minutes)
6. Go to the corresponding Notification to be sure ${logging_alert.messages_url} is part of the body template. If not, just add it
7. Wait 10 minutes the rule triggers

Check the value corresponding to ${logging_alert.messages_url} in the generated alert log.
In this URL we can see the "from" parameter and the "to" parameter which refer to the search timerange of the alert.
The difference between these two fields is only 1 minute but it should be 10 minutes (the timrange condition set in the rule).

It works well with a "MORE THAN 0" rule.

**Comment by c8y3** [comment]
When the event corresponding to rule LESS THAN is triggered it has no messages in the backlog. When this happens (or options singleMessage is set) the url is computed in a totally different way. I don't know why...

This can be reproduced without the wizard too:
* create a logging notification with template
   ```
   url: ${logging_alert.messages_url}
   subject: ${event_definition_title}
   ```
* create an aggregation event definition with condition count() < 1, search within 5 and which uses the previously declared notification, with a backlog of 50
* wait for 5 minutes and check the time range difference in the messages url is incorrectly 1 minute
* now change the condition to count() > 1
* send several logs to graylog (for instance as gelf inputs)
* wait for 5 minutes and check the time range difference is correct

**Comment by c8y3** [comment]
Ok, I cleaned up the code which computes the value of messages_url, while keeping the spirit of the previous implementation (I didn't want to break anything). This seems to solve the issue.
But, now that I understand this piece of code, I have many remarks about it. And I wonder if we just shouldn't be doing things in another, lots easier way.
Here is how the current implementation works to compute the time range in the url:
* for the start, just take the event eventTimestamp field (https://github.com/airbus-cyber/graylog-plugin-logging-alert/blob/6020ce5ed9fa33b676451fa220e7651a9d831cfb/src/main/java/com/airbus_cyber_security/graylog/events/notifications/types/LoggingAlert.java#L72)
* for the end, take the eventTimesamp+1, if the jobTrigger is not present, otherwise take the jobTrigger endTime+1, if present, otherwise take the jobTrigger triggeredAt+1. Then, if the end is before the start (I don't understand when this could happen), then take the start+1

This is complex and messy. And it doesn't give the same time range as the one presented in the alert detail under the "Aggregation time range" (see /alerts?page=1&pageSize=10 on a graylog with such alerts). By the way, the "Aggregation time range" is simply made up of the event timerangeStart and timerangeEnd fields.

Even worse, there seems to be a replay search link, which is simply /alerts/{event_identifier}/replay-search. 
Instead of making it very complex, shouldn't we just simply return this url?

**Comment by frantz45** [comment]
We can use the Aggregation time range, it seems to be exactly what we need.

Replay search is also a very good candidate to fill the whole messages_url, but:
- It doesn't work with the Correlation Count plugin
- "Group-by fields" need to be always set in order to find the right logs (to filter the search). Warning: currently with use the notification split fields to split/aggregate alerts but Graylog use the Event Definition group-by fields. Maybe we should stop setting split fields and only use group-by fields as the feature now exist in Graylog. (https://github.com/airbus-cyber/graylog-plugin-alert-wizard/issues/101)
- Currently the messages_url is used in SOAR. So be careful it has some impact
- The replay search only shows logs corresponding to the last triggered period. It doesn't show the whole period in case a notification is aggregated (linked to https://github.com/airbus-cyber/graylog-plugin-logging-alert/issues/5)

So for this issue you can just use Aggregation time range, but I think we need a deeper discussion to be more integrated to Graylog. Most of our developpements were made because many things were missing in Graylog, but now Graylog has really improved so we need to use builtin Graylog features

**Comment by c8y3** [comment]
Should be fixed in graylog-plugin-logging-alert 5.1.1. I still need to update the dependency in graylog-plugin-alert-wizard (and update changelog).
